reconcile/ocm_upgrade_scheduler.py

```diff
@@ -177,12 +172,72 @@ def version_conditions_met(version, history, ocm_name, workloads, upgrade_condit
     return conditions_met
 
 
+def gates_to_agree(version_prefix: str, cluster: str, ocm: OCM) -> list[str]:
+    """Check via OCM if a version is agreed
+
+    Args:
+        version_prefix (string): major.minor version prefix
+        cluster (string)
+        ocm (OCM): used to fetch infos from OCM
+
+    Returns:
+        bool: true on missing agreement
+    """
+
+    need_agreement = []
+    gate_ids = [gate["id"] for gate in ocm.get_version_gates(version_prefix)]
+    agreements = [
+        agreement["version_gate"]["id"]
+        for agreement in ocm.get_version_agreement(cluster)
+    ]
+
+    for gate in gate_ids:
+        if gate not in agreements:
+            need_agreement.append(gate)
+
+    return need_agreement
+
+
+def gate_agreeable(
+    version_agreements: Optional[list[str]], version_prefix: str
+) -> bool:
+    """Check, that a gate is configured to be agreed by this integration
+
+    Args:
+        version_agreements (list): strings representing agreeable version, * allows all
+        version_prefix (string): major.minor version prefix
+
+    Returns:
+        bool: True if conditions met
+    """
+    if version_agreements is None:
+        return False
+
+    for version in version_agreements:
+        if version == version_prefix:
+            return True
+        if version == "*":
+            return True
```

Comment:
`return any(v in (version_prefix, "*") for v in version_agreements)`

is shorter and more readable IMO.